File: trimixcodechallenge/src/main/java/com/trimix/challenge/service/PersonaService.java
package com.trimix.challenge.service;

import java.util.List;

import com.trimix.challenge.entity.Persona;

public interface PersonaService {
	
	public List <Persona> getPersonas();

    public void savePersona(Persona persona);

    public Persona getPersona(int theId);

    public void deletePersona(int theId);
    
    public List <Persona> getByNombreAndTipoDoc(String nombre, String tipoDoc);
    
    public List <Persona> getPersonaByTipoDoc(String tipoDoc);
}

File: trimixcodechallengeweb/src/app/service/persona.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {Persona} from "../model/persona.model";

@Injectable()
export class PersonaService {
    constructor(private http: HttpClient) {}
    baseUrl: string = 'http://localhost:8383/trimixcodechallenge/rest/persona';

    getPersonas() {
        return this.http.get<Persona[]>(this.baseUrl + '/list');
    }

    getPersonaById(id: number) {
        return this.http.get<Persona>(this.baseUrl + '/getById?personaId=' + id);
    }

    createPersona(persona: Persona) {
        return this.http.post(this.baseUrl + '/savePersona', persona);
    }

    updatePersona(persona: Persona) {
        return this.http.put(this.baseUrl + '/' + persona.id, persona);
    }

    deletePersona(id: number) {
        return this.http.delete(this.baseUrl + '/delete?personaId=' + id);
    }

    getPesonaByNombreAndTipoDocumento(persona: Persona){
        return this.http.get<Persona[]>(this.baseUrl + '/getByNombreAndTipoDoc?nombre=' + persona.nombre + '&tipoDoc=' + persona.tipoDocumento);
    }
}
